### PR TITLE
azure-cli-extensions.ssh: init at 2.0.5

### PR DESCRIPTION
**Pull request by Gordon-BP**
## Description of changes

1. Updated the version of all automatically-generated extensions by running the provided nix command
2. Manually added the [Azure CLI SSH Extension](https://learn.microsoft.com/en-us/cli/azure/ssh?view=azure-cli-latest) 
3. The ssh extension depends on [oschmod](https://pypi.org/project/oschmod/) which did not have a nixpkg; I made one and added it.
4. Added myself to the list of maintainers

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Gordon-BP** [comment]
@techknowlogick you were so helpful in my last PR, would you mind taking a look at this one please?

**Comment by katexochen** [comment]
Hi @Gordon-BP, thanks for you contribution! Some points regarding the commit structure:

1. Adding yourself as maintainer should be done in a separate commit, this should be the first commit of this PR. Commit title should be `maintainers: add gordon-bp`
2. Add the oschmod dependency next. There is a colon missing in the commit title.
3. Then add the extension you want to package, commit title `azure-cli-extensions.ssh: init at 2.0.5`

Regarding the update of generated extensions: this has already be done for 2.63.0. Of cause, the extensions get updated continuously, but I don't think we need to update these that often without a concrete reason. Each extension still needs to be updated in a separate commit with a correct commit title. There is a helper script in the package dir, but it usually also requires some manual work.

File: pkgs/development/python-modules/oschmod/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-vsmSFvMWFe5lOypch8rPtOS2GEwOn3HaGGMA2srpdPM=";
+  };
+  build-system = [ setuptools ];
+
+  meta = {
+    description = "oschmod is a Python package to change file permissions on Windows, macOS, and Linux";
```
Comment (katexochen) [inline pkgs/development/python-modules/oschmod/default.nix:20]:
```suggestion
    description = "Change file permissions on Windows, macOS, and Linux";
```

**Comment by Gordon-BP** [comment]
Hey @katexochen thanks for the review! Looks like I've got some more work to do on the commits - here is what I'm thinking:
1.Roll back all changes
2. Start with a maintainer add, commit with the message "maintainers: add gordon-bp"
3. Add only the `ssh` plugin to `extensions-manual.nix` but do not commit yet
4. Add the python package requirements for the ssh extension to `python-packages.nix`, THEN commit with the message "azure-cli-extensions.ssh: init at 2.0.5"
5. Open the PR and call it "azure-cli-extensions.ssh: init at 2.0.5"

However, since there is no nixpkg for the `oschmod` python package I still need to add it, but it would best be done like this?
1. Start a new branch cloned from `master`
2. Create a new `default.nix` for oschmod in `nixpkgs/pkgs/development/python-modules/oschmod` but don't commit yet.
3. Add oschmod to the list of python modules in `nixpkgs/pkgs/top-level/python-packages.nix` and then commit with the message "python3Packages.oschmod init at 0.3.12"
4. Open a new PR and call it "python3Packages.oschmod: init at 0.3.12"

Please let me know if I'm missing anything in my plan!

**Comment by katexochen** [comment]
> 1. Start a new branch cloned from `master`

No, it's totally fine to add `oschmod` in this PR, it should just be done before adding `azure-cli-extensions.ssh` (in the commit order). This way, each commit will be valid itself, while if you'd do it the other way (adding the extension first, then oschmod), the commit that adds the extension would be invalid, as it has the extension depending on a package that isn't existing at that point.

Otherwise, you plan sounds good. :) Remember that you can undo commits without loosing the changes by running 

```
git reset --soft HEAD~4
```

To sum up, expected commit history would look like this:

1. `maintainers: add gordon-bp`
2. `python3Packages.oschmod: init at 0.3.12`
3. `azure-cli-extensions.ssh: init at 2.0.5`

**Comment by Gordon-BP** [comment]
New commit order & messages

File: pkgs/by-name/az/azure-cli/extensions-manual.nix [hunk 2]
```diff
@@ -42,6 +42,20 @@
     ];
     meta.maintainers = with lib.maintainers; [ giggio ];
   };
+  ssh = mkAzExtension rec {
```
Comment (katexochen) [inline pkgs/by-name/az/azure-cli/extensions-manual.nix:45]: We try to keep this sorted alphabetically, so this should go underneath rdbms-connect.

File: pkgs/development/python-modules/msal/default.nix
Comment (katexochen) [inline pkgs/development/python-modules/msal/default.nix:1]: Why is the msal update needed?

File: pkgs/by-name/az/azure-cli/extensions-manual.nix [hunk 3]
```diff
@@ -42,6 +42,20 @@
...
+    version = "2.0.5";
+    url = "https://azcliprod.blob.core.windows.net/cli-extensions/ssh-${version}-py3-none-any.whl";
+    sha256 = "80c98b10d7bf1ce4005b7694aedd05c47355456775ba6125308be65fb0fefc93";
+    description = "SSH into Azure VMs using RBAC and AAD OpenSSH Certificates";
+    propagatedBuildInputs =
+      (with python3Packages; [
+        oras
+        oschmod
+      ])
+      ++ [ mycli ];
```
Comment (katexochen) [inline pkgs/by-name/az/azure-cli/extensions-manual.nix:56]: Is `mycli` needed? The extension doesn't seem to require it.

**Review by katexochen — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/msal/default.nix
Comment (katexochen) [reply pkgs/development/python-modules/msal/default.nix:1]: Ah, found https://github.com/NixOS/nixpkgs/issues/335750. There is already a PR for fixing this.

Comment (Gordon-BP) [reply pkgs/development/python-modules/msal/default.nix:1]: OK I will yeet the part about the msal upgrade :+1:

File: pkgs/by-name/az/azure-cli/extensions-manual.nix (hunk 3 above)
Comment (Gordon-BP) [reply pkgs/by-name/az/azure-cli/extensions-manual.nix:56]: I think it is left over from another manual extension which I copied from. Thanks for pointing it out!

**Comment by Gordon-BP** [comment]
ohhhh no I fucked this up real bad.... gonna nuke this PR and start another one sorry
